### PR TITLE
Improve readability on BTC recovery address

### PR DESCRIPTION
**Pull request by r-czajkowski**
Closes: #436

BTC Recovery Addresses need to be P2PKH or P2WPKH - this is a clunky and technical explanation both in the error message and on the How it Works page.

### CODE REVIEW

File: src/utils/forms.ts [hunk 1]
```diff
@@ -84,9 +84,12 @@ export const validateBTCAddress = (
...
-  } else if (!isValidBtcAddress(address, network)) {
-    return "Invalid btc address."
-  } else if (!isPublicKeyHashTypeAddress(address)) {
-    return "Must be a P2PKH or P2WPKH address."
+  } else if (
+    !isValidBtcAddress(address, network) ||
+    !isPublicKeyHashTypeAddress(address)
+  ) {
+    return `The BTC Recovery address has to start with the letters "${
+      network === Network.mainnet ? "bc" : "t"
```
Comment (michalsmiarowski) [inline src/utils/forms.ts:92]: Elaborating on my thought from the post above:

This address (`mipcBbFg9gMiCh81Kj8tqqdgoZub1ZJRfn`) that i took from https://en.bitcoin.it/wiki/List_of_address_prefixes passes the validation check on testnet and doesn't start with `t`

**Review by michalsmiarowski — CHANGES_REQUESTED** [review]
Left some comments to look at before the merge

File: src/pages/tBTC/Bridge/MintingCard/ProvideData.tsx [hunk 2]
```diff
@@ -31,6 +32,8 @@ type ComponentProps = {
   formId: string
 }
 
+const resolvedBTCAddressPrefix = supportedChainId === "1" ? "bc" : "t"
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/MintingCard/ProvideData.tsx:35]: cc @sashatanase

File: src/utils/forms.ts (hunk 1 above)
Comment (r-czajkowski) [reply src/utils/forms.ts]: 519f040

File: src/pages/tBTC/Bridge/MintingCard/ProvideData.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply src/pages/tBTC/Bridge/MintingCard/ProvideData.tsx]: 519f040

**Review by michalsmiarowski — APPROVED** [review]
LGTM 🥼

**Pull request merged**